[docs] Rename Expo client to Expo Go (#11747)
expo/expo: docs/pages/workflow/how-expo-works.md

The `sdkVersion` of an Expo app indicates what version of the compiled ObjC/Java/C layer of Expo to use. Each `sdkVersion` roughly corresponds to a release of React Native plus the Expo libraries in the SDK section of these docs.

> the Expo Go app supports many versions of the Expo SDK, but an app can only use one at a time. This allows you to publish your app today and still have it work a year from now without any changes, even if we have completely revamped or removed an API your app depends on in a new version. This is possible because your app will always be running against the same compiled code as the day that you published it.

Reviewer: ```suggestion
Expo Go supports many versions of the Expo SDK, but a project can only use one at a time. This allows you to publish your project today and still have it work a year from now without any changes, even if we have completely revamped or removed an API your project depends on in a new version. This is possible because your project will always be running against the same compiled code as the day that you published it.
```

> If you publish an update to your app with a new `sdkVersion`, if a user has yet to update to the latest Expo Go then they will still be able to use the previous `sdkVersion`.

Reviewer: ```suggestion
If you publish an update to your app with a new `sdkVersion`, if a user has yet to update to the latest version of Expo Go then they will still be able to use the previous `sdkVersion`.
```

## Opening Expo Apps Offline

> the Expo Go app will automatically cache the most recent version of every app it has opened. When you try to open an Expo app, it will always try and fetch the latest version, but if that fails for whatever reason (including being totally offline) then it will load the most recent cached version.

Reviewer: ```suggestion
Expo Go will automatically cache the most recent update of every project it has opened. When you try to open an Expo project, it will always try and fetch the latest update, but if that fails for whatever reason (including being totally offline) then it will load the most recent cached update.
```
